chore: update install instructions for Ibis 7.x

gforsyth opened the pull request:
Initial polish run to update things for changes in 7.x

The main change here is removing `relabel` in favor of `rename`, plus changing version specifiers in the installation instructions.

- chore: update install instructions for Ibis 7.x
- chore: update notebook 2 for ibis 7.x
- chore: add missing plotnine import

ncclementi commented on 01 - Getting Started.ipynb:
```diff
@@ -24,13 +24,13 @@
     "### `conda` / `mamba`\n",
```
<p>This import does not match what's on the readme, where you have</p><p> </p><p>#### conda / mamba</p><p> </p><p><code></code><code>sh</p><p>mamba create -n ibis-tutorial python ibis-duckdb=7 python-duckdb=0.8.1 jupyterlab altair plotnine</p><p></code><code></code></p><p> </p><p>#### pip</p><p> </p><p><code></code><code>sh</p><p>python -m pip install 'ibis-framework[duckdb]==7' duckdb==0.8.1 jupyterlab altair&gt;=5.0.1 plotnine</p><p></code><code></code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267019'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

ncclementi replied:
<p>When I run this, it created a temp file <code>.ddb.wal</code>, I would add that to the <code>.gitignore</code> and if worth it mention that is a temp file and not to worry about it for the purpose of this tutorial.    </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267023'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

ncclementi commented on 01 - Getting Started.ipynb:
```diff
@@ -24,13 +24,13 @@
     "### `conda` / `mamba`\n",
```
<p>I found helpful to see in the video version, the sql that is run. Maybe add a cell with</p><pre class="ql-syntax" spellcheck="false">ibis.show_sql(
	penguins.filter(
&nbsp;&nbsp;&nbsp;		[
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;	penguins.island == "Torgersen",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;	penguins.species == "Adelie",
&nbsp;&nbsp;&nbsp;		]
)) 
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267029'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

gforsyth replied:
<p>good idea!</p>

ncclementi commented on 01 - Getting Started.ipynb:
```diff
@@ -24,13 +24,13 @@
     "### `conda` / `mamba`\n",
```
<p>We haven't mention <code>.drop</code> before the exercise, I think it's probably fine, but just FYI.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267033'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

gforsyth replied:
<p>yeah, I'll leave this for now, but I can mention during the tutorial that <code>drop</code> does what you would expect it does </p>

ncclementi commented on 01 - Getting Started.ipynb:
```diff
@@ -24,13 +24,13 @@
     "### `conda` / `mamba`\n",
```
<p>Regarding the two solutions, is there a preferred way  (performance wise or best practice) between <code>.mutate</code> and <code>.drop</code> and <code>.select</code> . Maybe add a note about it?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267038'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

gforsyth replied:
<p>Good call -- using this as an opportunity to mention that the SQL we write and whatever the engine executes can be different things</p>

ncclementi commented on 01 - Getting Started.ipynb:
```diff
@@ -24,13 +24,13 @@
     "### `conda` / `mamba`\n",
```
<p>This solution uses </p><p> </p><pre class="ql-syntax" spellcheck="false">&nbsp;&nbsp;&nbsp; .group_by("island")
&nbsp;&nbsp;&nbsp; .body_mass_g.max()
</pre><p> </p><p>which is fine, but it's a slightly different syntax of what you just introduced which is </p><p> </p><pre class="ql-syntax" spellcheck="false">.group_by("island").agg(penguins.body_mass_g.max())
</pre><p> </p><p>Maybe you can mention both. It's just a nit pick</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267048'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

gforsyth replied:
<p>Good catch -- I'll change the solution to match the explanation for now.  I don't want to flood in too many options at the start.</p><p> </p>

ncclementi commented on 02 - Memtables, Joins, and Selectors.ipynb:
```diff
@@ -463,7 +463,7 @@
    "outputs": [],
```
<p>This is more of a question, since now in pandas you can have pyarrow dtypes, can you read or cast the "object" typo as a pyarrow dtype? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267057'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

gforsyth replied:
<p>Hmm, I don't think we've really dug into handling the pyarrow-backed pandas dataframes.</p><p> </p>

ncclementi commented on 02 - Memtables, Joins, and Selectors.ipynb:
```diff
@@ -463,7 +463,7 @@
    "outputs": [],
```
<p>not sure why this is here but you already have this above <code>t2 = ibis.memtable(pat)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267062'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

gforsyth replied:
<p>Oops!</p>

ncclementi commented on 02 - Memtables, Joins, and Selectors.ipynb:
```diff
@@ -463,7 +463,7 @@
    "outputs": [],
```
<p>This example is a bit confusing since this condition is always FALSE, maybe adding a comment saying that this performs a FULL JOIN  that returns all rows in both tables, using NULLs for values with no match.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267065'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

ncclementi replied:
<p>The solution has an extra column in the <code>.select()</code> it includes "isAdult" which is not part of the statement. But the following exercises need it, so the statement need to be fixed to include "isAdult" </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267079'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='70'/>

ncclementi replied:
<p>This should say "Using the <code>Table.rename</code> method "</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267085'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='71'/>

ncclementi replied:
<p>Are these tables in preload in snowflake by default, or you pull them in to your snowflake account? Might be worth leaving a comment. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267090'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='82'/>

ncclementi replied:
<p>might want to put a screenshot of the actual result so those that can't run this can still see the result. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267094'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='86'/>

ncclementi replied:
<p>I had to restart the notebook server to be able to run this. Restarting just the kernel didn't work. </p><p> </p><p>Not sure if this is a bug, or what, but I had the same issue you did on you EuroScipy tutorial (saw recording)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267099'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='91'/>

ncclementi replied:
<p>After this, do you want to do thge same than in notebook 1 <code>del con</code> ?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267104'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='110'/>

ncclementi commented on 02 - Memtables, Joins, and Selectors.ipynb:
```diff
@@ -463,7 +463,7 @@
    "outputs": [],
```
<ol><li>Is there a reason why we use a dictionary here? In the docs examples <a href="https://ibis-project.org/reference/expression-tables#ibis.expr.types.relations.Table.rename" target="_blank">here</a> we go over multiple ways of renaming columns, but there is no dictionary like example. Why not just doing `.r<code>.rename(new_name="old_name")</code></li><li>Maybe used the word "renamed" in the comments instead of relabeled just to be consistent.</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267071'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

gforsyth replied:
<p>nope, just a sloppy update from <code>relabel</code> to <code>rename</code> -- good catch on the language, too  </p>

ncclementi commented on 02 - Memtables, Joins, and Selectors.ipynb:
```diff
@@ -463,7 +463,7 @@
    "outputs": [],
```
<p>What's the difference between this <code>basics</code> and the one two cells above. Not super clear. Do we need this step?</p><p> </p><p>Same question for ratings</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267076'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>

gforsyth replied:
<p>they're the same table -- I'll add a note</p>

ncclementi commented on 02 - Memtables, Joins, and Selectors.ipynb:
```diff
@@ -463,7 +463,7 @@
    "outputs": [],
```
<p>add <code>from ibis import _</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267101'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='98'/>

gforsyth replied:
<p>It's earlier in the notebook, but yeah, I'll add it in again because of needing to restart the notebook server (and I'll figure out how to avoid that...)</p>

ncclementi commented on 03 - Playing with PyPI.ipynb:
```diff
@@ -388,7 +388,7 @@
    "metadata": {},
```
<p>This one took couple of seconds for me to load, not sure if that's ok but just as an FYI</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ibis-project/ibis-tutorial/pull/3/discussion/#comment-1361267108'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

ncclementi commented:
Hi @gforsyth I left some comments, but overall this looks great. and it was quite straightforward to follow. I ran into some issues that I put on the comments, but having the video recording from EuroScipy was helpful (you can add it to the readme as previous iterations of this tutorial).

I see that the time allocated for this for pydata-nyc is 90 min I wonder if the last notebook instead of doing all of it as exercises, you do it more like a demo.

gforsyth commented:
Thanks for the thorough review @ncclementi !  I've addressed most of your comments (I think).  It's definitely too much material for 90 minutes -- I'm going to aim to get through the first two notebooks in 90 minutes and leave the 3rd notebook as "homework" if people want to explore further.

ncclementi commented:
Thanks @gforsyth This LGTM now. 

I only found two small thing:

1. The README is pointing to `git clone https://github.com/gforsyth/ibis-tutorial.git`, you probably want to point to `git clone https://github.com/ibis-project/ibis-tutorial.git`
2. Nitpick: at the end of notebook 2 we are not closing the connections, as suggested in notebook 1.

gforsyth commented:
> 1. The README is pointing to `git clone https://github.com/gforsyth/ibis-tutorial.git`, you probably want to point to `git clone https://github.com/ibis-project/ibis-tutorial.git`

good catch!



> Nitpick: at the end of notebook 2 we are not closing the connections, as suggested in notebook 1.

I'm going to ignore this for now, until I can confirm that it actually does what we want it to do (which is remove the lock on the penguins ddb file)
